src/js/search-bar/search-bar.view.js

```diff
@@ -4,26 +4,44 @@ import {selectDomElement} from "../utils/dom-selector";
 import {SearchBarEvent} from "./search-bar.constants";
 
 const template = `
-    <button id="awesome-button">click</button>
-`
+    <div class="search-bar hover-initial-search-bar" id="search-bar">
+        <input class="search-bar__input" id="search-input" type="search" placeholder="Search...">
+        <button class ="search-bar__icon" id="search-btn">
+            <img src="https://i.ibb.co/d5rcTzD/icons8-magnifying-glass-64.png" alt="search-icon" class ="search-bar__img">
+        </button>
+    <div class="search-bar__overlay" id="search-bar-overlay"></div>
+    </div>
+`;
 
 export class SearchBarView extends AbstractView {
 
-    awesomeButton;
+    searchButton;
 
     onViewInit() {
         super.onViewInit();
-        this.selector = '#search-bar';
+        this.selector = '#search-system';
         this.template = template;
     }
 
     afterViewInit() {
         super.afterViewInit();
-        this.awesomeButton = selectDomElement('#awesome-button');
-        this.awesomeButton.onclick = this.sendAwesomeEvent.bind(this);
+        this.searchBarOverlay = selectDomElement('#search-bar-overlay');
+        this.searchBtn = selectDomElement('#search-btn');
+
+        document.onkeyup = (event) => {
+            if (event.key === 'Enter') {
+                this.sendEvent(SearchBarEvent.CLICK_SEARCH);
+            };
+        };
+        this.searchBarOverlay.onclick = () => { 
+            this.sendEvent(SearchBarEvent.CLICK_SEARCHBAR_OVERLAY);
```

Comment:
please care about naming. previous one was just an example for dummy button click. It is better to use a verb of an action, like toggle, send etc, but not just click